Changes to support ci_build/install scripts with Ubuntu 18.04 docker …

wdirons opened the pull request:
…containers

Switched from clang-format-3.8 to clang-format-3.9, as it is available on
Ubuntu 18.04, in addition to being available on Ubuntu 14.04 and Ubuntu 16.04.

python-setuptools in Ubuntu 18.04 no longer includes easy_install. For Ubuntu
18.04 only, installing pip using the https://bootstrap.pypa.io/get-pip.py
python program. Ubuntu 16.04 and 14.04 behavior unchanged.

Force installing lazy-object-proxy version 1.4.1, because non x86_64 systems
that need to compile from source encounter an error with version 1.4.2:
`AttributeError: module 'setuptools.build_meta' has no attribute '__legacy__'`
This error is because we install setuptools with apt, and then replace
setuptools with a newer version using pip. This error is resolved in the latest
version of setuptools.

wdirons committed:
Changes to support ci_build/install scripts with Ubuntu 18.04 docker containers

Switched from clang-format-3.8 to clang-format-3.9, as it is available on
Ubuntu 18.04, in addition to being available on Ubuntu 14.04 and Ubuntu 16.04.

python-setuptools in Ubuntu 18.04 no longer includes easy_install. For Ubuntu
18.04 only, installing pip using the https://bootstrap.pypa.io/get-pip.py
python program. Ubuntu 16.04 and 14.04 behavior unchanged.

Force installing lazy-object-proxy version 1.4.1, because non x86_64 systems
that need to compile from source encounter an error with version 1.4.2:
AttributeError: module 'setuptools.build_meta' has no attribute '__legacy__'
This error is because we install setuptools with apt, and then replace
setuptools with a newer version using pip. This error is resolved in the latest
version of setuptools.

wdirons committed:
Install future with quotes to handle the > symbol

pip install future>=0.17.1 will run the command `pip install future`
and redirect the output to a file called `=0.17.1`. By putting the command
in quotes the redirection is avoided and proper command is run.

r4nt commented:
Looks good from my side, thanks for prepping this for 18!

gunan commented on tensorflow/tools/ci_build/install/install_deb_packages.sh:
```diff
@@ -43,7 +43,7 @@ apt-get install -y --no-install-recommends \
     automake \
     build-essential \
-    clang-format-3.8 \
+    clang-format-3.9 \
```
Is this package necessary?
I feel like it may be a leftover package.

wdirons replied:
I will try some test on x86 without this package and verify it works ok.

gunan commented on tensorflow/tools/ci_build/install/install_pip_packages.sh:
```diff
@@ -15,11 +15,20 @@
-
+# easy_install is not available on ubuntu 18.04
+if [[ "$ubuntu_version" == "18" ]]; then
+  wget https://bootstrap.pypa.io/get-pip.py
```
Should we just switch all ubuntu versions to this?
This is the official installation method anyway.

That way, we can avoid version specific checks.

wdirons replied:
I was trying to avoid changing existing behavior as much as possible, but I can test this on x86 and verify it works and then push an update to the PR.

wdirons committed:
Changes to support ci_build/install scripts with Ubuntu 18.04 containers

Updates for code review comments:

* Removed clang-format from being installed with apt. It is not longer
needed.

* Removed the Ubuntu version check for installing pip, always download
and run the get-pip.py script to install pip for all versions of Ubuntu

wdirons commented:
I made the changed requested. I tested docker build on x86 with the following files:
Dockerfile.rbe.ubuntu16.04-manylinux2010 (Ubuntu 16.04)
Dockerfile.custom_op_gpu (Ubuntu 14.04)

Tested bazel build using a container created from the first image.
